src/WebJobs.Script/Description/PowerShell/PowerShellFunctionInvoker.cs

```diff
@@ -0,0 +1,490 @@
+
+                    PSDataCollection<PSObject> outputCollection = new PSDataCollection<PSObject>();
+                    outputCollection.DataAdded += OutputCollectionDataAdded;
+
+                    powerShellInstance.Streams.Error.DataAdded += ErrorDataAdded;
+
+                    IAsyncResult result = powerShellInstance.BeginInvoke<PSObject, PSObject>(null, outputCollection);
+
+                    while (result.IsCompleted == false)
+                    {
+                        System.Threading.Thread.Sleep(50);
+                    }
+
+                    // Clean up and remove module
+                    if (_moduleFiles.Any())
+                    {
+                        powerShellInstance.AddCommand("Remove-Module").AddArgument(_moduleFiles);
+                    }
+                }
+
+                runspace.Close();
+            }
+        }
+
+        private void LogLoadedModules()
+        {
+            List<string> moduleRelativePaths = new List<string>();
+            foreach (string moduleFile in _moduleFiles)
+            {
+                string relativePath = GetRelativePath(moduleFile);
+                moduleRelativePaths.Add(relativePath);
+            }
+
+            if (moduleRelativePaths.Any())
+            {
+                TraceWriter.Info(string.Format("Loaded modules:{0}{1}", Environment.NewLine, string.Join(Environment.NewLine, moduleRelativePaths)));
+            }
+        }
+
+        private string GetRelativePath(string moduleFile)
+        {
+            string pattern = string.Format("^.*?(?=\\\\{0}\\\\)", _functionName);
+            MatchCollection matchCollection = Regex.Matches(moduleFile, pattern);
+            string newtoken = moduleFile.Replace(matchCollection[0].Value, string.Empty);
+            string relativePath = newtoken.Replace('\\', '/');
+            return relativePath;
+        }
+
+        private static void SetProcessEnvironmentVariables(IDictionary<string, string> envVariables)
+        {
+            foreach (var pair in envVariables)
+            {
+                Environment.SetEnvironmentVariable(pair.Key, pair.Value, EnvironmentVariableTarget.Process);
+            }
+        }
+
+        private static void DeleteProcessEnvironmentVariables(IDictionary<string, string> envVariables)
+        {
+            foreach (var pair in envVariables)
+            {
+                var envSetting = Environment.GetEnvironmentVariable(pair.Key, EnvironmentVariableTarget.Process);
+                if (!string.IsNullOrEmpty(envSetting))
+                {
+                    Environment.SetEnvironmentVariable(pair.Key, null, EnvironmentVariableTarget.Process);
+                }
+            }
+        }
+
+        private async Task ProcessInputBindingsAsync(object input, string functionInstanceOutputPath, IBinderEx binder, Dictionary<string, string> bindingData, Dictionary<string, string> environmentVariables)
+        {
```

Comment:
Yeah, this will be a big maintenace headache. We need to refactor this into a shared base class so you can just inherit it.